SimplenoteWidgets/Views/NoteListTable.swift

```diff
@@ -0,0 +1,31 @@
+import SwiftUI
+
+struct NoteListTable: View {
+    let entry: ListWidgetEntry
```

Comment:
The widget provider is released after the timeline is created, which release the core data manager and subsequently all of the managed objects.  To pass in the actual items we would also need to pass in the MOC.  I though it would be simpler to pass in the content needed to the widgets instead of maintaining the connection to core data

Reply:
Right, what I meant is that the only thing `NoteListTable` does with `ListWidgetEntry` is `entry.noteProxys`, so why not pass `entry.noteProxys` there is the first place.

If we combine this with the requirement to fill the whole table with rows without leaving empty space I suggest to represent each row as an enum with 2 cases: `note(Proxy)` and `empty`. I think this will better show the intention.

```
enum Row {
    case note(ListWidgetNoteProxy)
    case empty
}

var rows: [Row] {
    let numberOfRows = rows(for: widgetFamily)
    var data = entry.noteProxys.map({ Row.note($0) })
    for _ in data.count..<numberOfRows {
        data.append(.empty)
    }
    return data
}

NoteListTable(rows: rows,
              geometry: geometry)
```

Reply:
I like it a lot!  I will make this change!